src/Host.cpp

```diff
@@ -1735,21 +1735,17 @@ void Host::setUserDictionaryOptions(const bool _useDictionary, const bool useSha
 // however it should ensure that other classes get updated:
 void Host::setName(const QString& newName)
 {
+    if (mHostName == newName) {
+        return;
+    }
+
     int currentPlayerRoom;
     if (mpMap) {
         currentPlayerRoom = mpMap->mRoomIdHash.take(mHostName);
     }
 
     QMutexLocker locker(& mLock);
```

Comment:
I think there are a couple of things that should be done differently - `currentPlayerRoom` should be initialised to `0` and only reinserted if it is set to something else (`QMake::take(const T&)` will return `0` if the value is not found) so the method code should be:
```
{
    if (mHostName == newName) {
        return;
    }

    int currentPlayerRoom = 0;
    if (mpMap) {
        currentPlayerRoom = mpMap->mRoomIdHash.take(mHostName);
    }

    QMutexLocker locker(& mLock);
    // Now we have the exclusive lock on this class's protected members
    mHostName = newName;
    // We have made the change to the protected aspects of this class so can unlock the mutex locker and proceed:
    locker.unlock();

    mTelnet.mProfileName = newName;
    if (mpMap) {
        mpMap->mProfileName = newName;
        if (currentPlayerRoom) {
            mpMap->mRoomIdHash.insert(newName, currentPlayerRoom);
        }
    }

    if (mpConsole) {
        mpConsole->setProfileName(newName);
    }
    mTimerUnit.changeHostName(newName);
}

Reply:
Adjusted.